<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MakePrimUI</class>
 <widget class="QDialog" name="MakePrimUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>336</width>
    <height>147</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate a Primitive</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="gb_type">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Type</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="1" column="0">
           <widget class="QRadioButton" name="rb_circle">
            <property name="text">
             <string>Circle</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <property name="autoRepeat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QRadioButton" name="rb_cylinder">
            <property name="text">
             <string>Cylinder</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QRadioButton" name="rb_disc">
            <property name="text">
             <string>Disc</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QRadioButton" name="rb_ndisc">
            <property name="text">
             <string>Disc Negative</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QRadioButton" name="rb_ring">
            <property name="text">
             <string>Ring</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QRadioButton" name="rb_cone">
            <property name="text">
             <string>Cone</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QCheckBox" name="cb_hires">
         <property name="text">
          <string>Hi-res</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Segments:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Ring number:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="sb_segs">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>16</number>
           </property>
           <property name="value">
            <number>16</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="sb_ringnum">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>MakePrimUI</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>MakePrimUI</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rb_circle</sender>
   <signal>clicked(bool)</signal>
   <receiver>sb_ringnum</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rb_cylinder</sender>
   <signal>clicked(bool)</signal>
   <receiver>sb_ringnum</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rb_disc</sender>
   <signal>clicked(bool)</signal>
   <receiver>sb_ringnum</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rb_ndisc</sender>
   <signal>clicked(bool)</signal>
   <receiver>sb_ringnum</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>83</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rb_ring</sender>
   <signal>clicked(bool)</signal>
   <receiver>sb_ringnum</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rb_cone</sender>
   <signal>clicked(bool)</signal>
   <receiver>sb_ringnum</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>111</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>enableRingNumber()</slot>
 </slots>
</ui>
